### PR TITLE
Issue #614: Introduce a dirty flag for variables.

### PR DESCRIPTION
**Pull request by jaypha**

### CODE REVIEW

File: classes/dataflow.php [hunk 1]
```diff
@@ -47,6 +47,12 @@ class dataflow extends persistent {
     /** @var bool Set to true when the dataflow is in the process of deleting. */
     private $isdeleting = false;
 
+    /** @var bool If true, then the variables tree will be rebuilt when get_variables() is called.  */
+    private $rebuildvariables = true;
```
Comment (keevan) [inline classes/dataflow.php:51]: This matches the method name which could get a bit messy.

Are we able to change this to be something else such as `shouldrebuildvariables` (and maybe the variable tree should be its own class instead - to hold its own internal state?)

**Review by keevan — COMMENTED** [review]
A minor rename comment, and a sugquestion.

File: classes/dataflow.php (hunk 1 above)
Comment (jaypha) [reply classes/dataflow.php:51]: Changing the variable name is fine.
Putting into it's own class? Well, yeah, that's kinda the point of the issue, but that is VERY complicated, and this is supposed to be a quick, quick fix.

**Review by keevan — APPROVED** [review]
There's a clear performance improvement from local testing, and the flows I've tested with seem to run as expected.

So approving this one.

**Pull request merged**